---
title: "Achiever Microbial Preliminary Analyses"
author: "Carolyn Prentice"
date: "30/03/2020"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

First we need to load the required packages

```{r Packages}

#load packages
library("phyloseq"); packageVersion("phyloseq")
library(tidyverse)
library(dbplyr)
library(extrafont)
library(viridis)
library(ggplot2)
library(vegan)
library(rmarkdown)
library(data.table)

```


# Introduction

In June of 2019, the Hakai Institute surveyed three of British Columbia's coastal fjords - Bute, Knight and Toba Inlets. The goal of the survey was to characterize the (micro)biology, chemistry, and physics of the water column of these glacially influenced fjords, where climate change impacts are expected to be acute and where changes in ocean conditions are already occuring.

In this file, we will examine the microbial (Bacteria, Archaea) and Phytoplankton communtiies in these three fjords, as well as from three rivers flowing into Bute Inlet, and seasonal trends at station QU43, which is located at the mouth of Bute Inlet and sampled all year round. Information on the microbial communities was obtained by extracting DNA from water samples and sequencing the V4-V5 region of the 16S rRNA gene. 


```{r Load Microbe Data}

#load 16S microbial taxonomy, asv, and metadata files
asvfile <-read.csv("Achiever_QU43_ASV_2020.csv", row.names=1, header = TRUE)
taxfile <-as.matrix(read.csv("Achiever_QU43_Taxa_2020.csv",row.names=1,header = TRUE))
mapfile <-read.csv("Achiever_QU43_Metadata_2020.csv", row.names=1,header=TRUE)

```

We will create a PhyloSeq object, which constists of the following:
1. An OTU (Operational Taxonomic Unit) Table, or in this case a Amplicon Sequence Varient (ASV) Table
2. A Taxa file
3. A Metadata file 

```{r Create PhyloSeq Object}
##turn them into components for phyloseq object

OTU1 = otu_table(asvfile, taxa_are_rows = TRUE)
TAX1 = tax_table(taxfile)
MAP1 = sample_data(mapfile)

#Make Phyloseq Object
Achiever_Bac = merge_phyloseq(OTU1, TAX1, MAP1)

ntaxa(Achiever_Bac)
#2814 taxa before removing singletons

```

Next we will remove any taxa that appear only once in the dataset 

```{r Remove Singletons}
#Remove singletons -- taxa that appear only once

Achiever_Bac<-prune_taxa(taxa_sums(Achiever_Bac) > 1, Achiever_Bac) 

ntaxa(Achiever_Bac)  #2804 taxa once singletons removed

```

And do some summary statistics of the data set

```{r Phyloseq Summary Statistics}

#gives us the number and taxa and number of samples
otu_table(Achiever_Bac) #2804 taxa and 107 samples
sample_data(Achiever_Bac)
tax_table(Achiever_Bac)


ntaxa(Achiever_Bac)  #2804
```

First we can get a sense of the data by looking at the various phyla present in the three inlets (top panel) compared with the River samples and station QU43 samples

```{r Bar Plot of Phyla across Locations}


#from Colleen's R code:

#First summarize data by phylum
achiever_phylum <- Achiever_Bac %>%
  tax_glom(taxrank = "Phylum") %>%                        # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} )    # Transform to rel. abundance


#Convert phyloseq object to a normal data table to make more customizable plots
ach_dat <- data.table(psmelt(achiever_phylum))

#Convert Phylum to a character vector from a factor because R
ach_dat$achiever_phylum <- as.character(ach_dat$achiever_phylum)

# group dataframe by Phylum, calculate median rel. abundance
ach_dat[, mean := mean(Abundance, na.rm = TRUE), 
    by = "Phylum"]

#Change name to remainder of Phylum less than .5%
ach_dat[(mean <= 0.005), Phylum := "Remainder"]

#Do this so that R will plot in this specific order
ach_dat$Depth2 <- factor(ach_dat$Depth2, levels = c("0","5","10","25-30","90","100-200","200-300","400-500",">500"))


#Do this so the 3 fjords will be grouped on top for the facet wrap using Location3
ach_dat$Location3 <- factor(ach_dat$Location3, levels = c("Bute", "Knight", "Toba", "QU43", "River"))


#Bar graph of dominant phyla faceted by locations (fjords, river, qu43)
ggplot(ach_dat, aes(x=Phylum, y=Abundance)) +
  geom_bar(aes(fill = Phylum), stat = "summary", fun.y = "mean", position = "dodge") +
  scale_fill_viridis_d(option="plasma") +
  theme_bw() +
  facet_wrap(~Location3) +
  theme(axis.text.x = element_text(angle = 35, hjust = 1), legend.position = "none", strip.text = element_text(size = 15)) +
  labs(x="Phylum", y="Average Relative Abundance")
  
```

You can see that some phyla dominate across all locations (**Proteobacteria**, **Bacteroidota**), while others are less prominent and vary slightly across locations (e.g. **Crenarchaeota** in marine locations, but absent from rivers, **Actinobacteriota** slightly more abundant in rivers than marine locations)


Next we can do the same thing but visualize the phyla by **depth** bins as well as Location

```{r Bar Plot of Phyla across Depths}

#Bar graph of dominant phyla faceted by depths
ggplot(ach_dat, aes(x=Phylum, y=Abundance)) +
  geom_bar(aes(fill = Phylum), stat = "summary", fun.y = "mean", position = "dodge") +
  scale_fill_viridis_d(option="plasma") +
  theme_bw() +
  facet_wrap(~Depth2) +
  theme(axis.text.x = element_text(angle = 35, hjust = 1), legend.position = "none", strip.text = element_text(size = 15)) +
  labs(x="Phylum", y="Average Relative Abundance")


```

Similar to the location trends, you can see that some phyla dominate across all depths (namely **Proteobacteria** and **Bacteroidota**). However, the **Bacteroidota** seem to become less abundant at greater depths. The diversity of phyla also seems to increase with depth, with a few phyla becoming apparently more abundant as depth increases (e.g. **Marine group B**).


Next, we can start to get a sense of if the community changes seasonally by looking at phyla across seasons. Note that only QU43 was sampled across the seasons so I will only look at QU43 for this

```{r Bar Plot of Phyla across Seasons}

#Bar graph of dominant phyla faceted by season

achiever_phylum_qu43 <- Achiever_Bac_QU43 %>%
  tax_glom(taxrank = "Phylum") %>%                        # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} )    # Transform to rel. abundance

#Convert phyloseq object to a normal data table to make more customizable plots
ach_dat_qu43 <- data.table(psmelt(achiever_phylum_qu43))

#Convert Phylum to a character vector from a factor because R
ach_dat_qu43$achiever_phylum_qu43 <- as.character(ach_dat_qu43$achiever_phylum_qu43)

# group dataframe by Phylum, calculate median rel. abundance
ach_dat_qu43[, mean := mean(Abundance, na.rm = TRUE), 
    by = "Phylum"]

#Change name to remainder of Phylum less than .5%
ach_dat_qu43[(mean <= 0.005), Phylum := "Remainder"]

ggplot(ach_dat_qu43, aes(x=Phylum, y=Abundance)) +
  geom_bar(aes(fill = Phylum), stat = "summary", fun.y = "mean", position = "dodge") +
  scale_fill_viridis_d(option="plasma") +
  theme_bw() +
  facet_wrap(~Season) +
  theme(axis.text.x = element_text(angle = 35, hjust = 1), legend.position = "none", strip.text = element_text(size = 15)) +
  labs(x="Phylum", y="Average Relative Abundance")


```

At least at the phylum level there aren't any strong trends among seasons


Here I will do some ordination plots using Colleen's code  

```{r ORDINATION 2}

set.seed(1)
achiever_nmds <- ordinate(
  physeq = Achiever_Bac, 
  method = "NMDS", 
  distance = "bray",
  trymax = 50
)


p1 <- plot_ordination(
  physeq = Achiever_Bac,
  ordination = achiever_nmds,
  color = "Location2", #you can change this variable to other things like "Depth", "Station", "Location" ,etc. and it will automatically change the colors.
  shape = "Depth2"
  ) 

#order Location2 in legend in a specific order, rather than alphabetical
p1$data$Location2 <- factor(p1$data$Location2, levels = c("4791","Homathko","QU43" ,"Bute","Toba","Knight"))

#Order Depth2 in legend in descending order
p1$data$Depth2 <- factor(p1$data$Depth2, levels = c("0","5","10","25-30","90","100-200","200-300","400-500",">500"))


#Changing aesthetics of plot1 -- colour, point size etc.


#now plot, using the viridis color scheme. can change to other or use specific colors, as above.
p1 + geom_point(size = 3) + 
  scale_color_brewer(palette = "Spectral") +
  theme_bw() +
  theme(
    panel.background = element_rect(fill = "transparent",colour = NA),
    plot.background = element_rect(fill = "transparent",colour = NA),
    panel.grid.major = element_blank(), 
    panel.grid.minor = element_blank())

p1


#scale_color_viridis(discrete=TRUE, option="plasma")

```



```{r PERMARNOVA}
#to perform a PERMANOVA

metadata <- as(sample_data(Achiever_Bac), "data.frame")

#seeing if there is a significant effect of Depth

adon_depth<-adonis(distance(Achiever_Bac, method="bray") ~ Depth2,
       data = metadata)
adon_depth

#effect of Location? (Toba, Knight, Bute, QU43, Homathko, 5005, Orford)

adon_loc<-adonis(distance(Achiever_Bac, method="bray") ~ Location2,data = metadata)

adon_loc

```


Here I am bringing in the chloroplast data from the 16S sequences

```{r Load Phytoplankton Data}
knitr::opts_chunk$set(echo = TRUE)

#load taxonomy, asv, and metadata files
asvfile2 <-read.csv("Achiever_QU43_ASV_2020_chloroplast.csv", row.names=1, header = TRUE)
taxfile2 <-as.matrix(read.csv("Achiever_QU43_Taxa_2020_chloroplast.csv",row.names=1,header = TRUE))
mapfile2 <-read.csv("Achiever_QU43_Metadata_2020_chloroplast.csv", row.names=1,header=TRUE)

```


## Including Plots

You can also embed plots, for example:

```{r plot, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
